CMakeModules/FindZ3.cmake

```diff
@@ -50,3 +51,20 @@ else()
     message(STATUS "Z3 includes directory defined: ${Z3_INCLUDE_DIRS}")
     message(STATUS "Z3 libraries defined: ${Z3_LIBRARIES}")
 endif()
+
+find_file(Z3_VERSION_HEADER
+  z3_version.h
+  PATHS ${Z3_PKGCONF_INCLUDE_DIRS}
```

Comment:
This variable may not be present when building with z3 in custom directory that is not installed system-wide.

Reply:
We may try searching in Z3_INCLUDE_DIRS if Z3_PKGCONF_INCLUDE_DIRS are not present.

Reply:
I guess I have the same question as in #1117 : I am curious about the workflow you described.
So far I am aware of `find_package`:
- using the `MODULE` mode which uses `FindZ3.cmake`
- using the `CONFIG` mode which will uses a `Z3Config.cmake`, and then the way to define `Z3_VERSION` is document through the CMake Config mode mechanism.